### ISSUE TITLE
Failed to download latest patch version (os error 111)

### ISSUE THREAD

**Issue by LMFuture-Me** [an-anime-team/the-honkers-railway-launcher#47]
OS: openSUSE Tumbleweed 20230624
Glibc: 2.37
[debug.log](https://github.com/an-anime-team/the-honkers-railway-launcher/files/11856346/debug.log)

**Comment by Mar0xy** [comment]
This happens if well the connection is being refused try to visit codeberg.org and see if it works

**Comment by fsl3118** [comment]
> This happens if well the connection is being refused try to visit codeberg.org and see if it works

 Codeberg.org seems to be normal, but the problem still exists.

**Comment by LMFuture-Me** [comment]
> This happens if well the connection is being refused try to visit codeberg.org and see if it works

yes. I can visit codeberg.org. I think it is a better idea to use curl or reqwest to do this.
